File: solutions/swift/pythagorean-triplet/1/Sources/PythagoreanTriplet/PythagoreanTriplet.swift
func tripletsWithSum(_ sum: Int) -> [[Int]] {
    (1...(sum - 1) / 3).compactMap { a -> [Int]? in
        let b = (sum * sum - 2 * sum * a) / (2 * sum - 2 * a)
        let c = sum - a - b

        return isPythagorean(a: a, b: b, c: c) ? [a, b, c] : nil
    }
}

private func isPythagorean(a: Int, b: Int, c: Int) -> Bool {
    (a < b) && (b < c) && (a * a + b * b == c * c)
}

File: solutions/vbnet/pythagorean-triplet/1/PythagoreanTriplet.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq

Public Module PythagoreanTriplet
    Public Iterator Function TripletsWithSum(ByVal sum As Integer) As IEnumerable(Of (a As Integer, b As Integer, c As Integer))
        For a As Integer = 1 To sum \ 3
            For b As Integer = a + 1 To sum \ 2
                If a * a + b * b = (sum - a - b) * (sum - a - b) Then
                    Yield (a, b, sum - a - b)
                End If
            Next
        Next
    End Function
End Module